// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package dbscanserving;

enum Metric {
    EUCLIDEAN = 0;
}

message Sample {
    repeated float features = 1;
}

message DetectionRequest {
    float eps = 1;
    int32 min_samples = 2;
    Metric metric = 3;
    int32 num_samples = 4;
    int32 num_features = 5;
    repeated Sample samples = 6;
    int32 identifier = 7;
}

message DetectionResponse {
    repeated int32 cluster_indices = 1;
}

service Detector {
    rpc Detect (DetectionRequest) returns (DetectionResponse);
}
